<a href="https://colab.research.google.com/github/juanignaciogiacobbe/7506R-2C2023-GRUPO15/blob/main/7506R_TP1_GRUPO15_CHP1_ENTREGA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <center>Organizaci&oacute;n de Datos</center>
#### <center>C&aacute;tedra Ing. Rodriguez, Juan Manuel </center>

## <center>Trabajo Práctico 1: Reservas de Hotel</center>
### <center> Grupo número 15</center>
### <center> Giacobbe, Juan Ignacio; alumno2; alumno3</center>


#Checkpoint 1: Análisis exploratorio y Preprocesamiento de Datos

Comenzamos importando todas las librerías que vamos a utilizar en el proyecto

In [6]:
import pandas as pd
import numpy as np

Cargamos los archivos csv de datos en dos DataFrames de Pandas

In [10]:
ds_hoteles_test = pd.read_csv('./7506R-2C2023-GRUPO15/hotels_test.csv')
ds_hoteles_train = pd.read_csv('./7506R-2C2023-GRUPO15/hotels_train.csv')

Vamos a empezar viendo qué forma tiene el dataset, cuántas variables tiene y qué describe cada campo, de esta forma podemos saber de antemano con qué nos enfrentamos.

Vemos información sobre las 2 primeras filas del DataFrame

In [19]:
ds_hoteles_test.head(2)

hotel  lead_time  arrival_date_year arrival_date_month  \
0  City Hotel        277               2016           November   
1  City Hotel        286               2016                May   

   arrival_date_week_number  arrival_date_day_of_month  \
0                        46                          7   
1                        20                         11   

   stays_in_weekend_nights  stays_in_week_nights  adults  children  ...  \
0                        1                     2       2       0.0  ...   
1                        0                     4       2       0.0  ...   

   booking_changes deposit_type agent company days_in_waiting_list  \
0                0   Non Refund   NaN     NaN                    0   
1                0   No Deposit  30.0     NaN                  176   

     customer_type    adr  required_car_parking_spaces  \
0        Transient  100.0                            0   
1  Transient-Party  101.0                            0   

  total_of_special_requests                                    id  
0                         0  e6d0d58a-ab96-44d3-b28f-85df1cc15f1e  
1                         0  aba9aac2-4495-4645-a60e-90328713c999  

[2 rows x 30 columns]

Vemos información de las últimas 2 filas del DataFrame

In [20]:
ds_hoteles_test.tail(2)

hotel  lead_time  arrival_date_year arrival_date_month  \
26533    City Hotel        164               2017             August   
26534  Resort Hotel        136               2016              March   

       arrival_date_week_number  arrival_date_day_of_month  \
26533                        32                         10   
26534                        13                         24   

       stays_in_weekend_nights  stays_in_week_nights  adults  children  ...  \
26533                        2                     4       2       0.0  ...   
26534                        0                     3       2       0.0  ...   

       booking_changes deposit_type  agent company days_in_waiting_list  \
26533                0   No Deposit   85.0     NaN                    0   
26534                0   No Deposit  387.0     NaN                    0   

         customer_type   adr  required_car_parking_spaces  \
26533        Transient  80.1                            0   
26534  Transient-Party  84.0                            0   

      total_of_special_requests                                    id  
26533                         0  89c85fa5-2b3d-4131-844c-e151a65ca240  
26534                         0  bd4e9eee-f28e-4a53-b4a1-277426efbcee  

[2 rows x 30 columns]

Vemos cuántas filas y cuántas columnas tiene el DataFrame

In [13]:
ds_hoteles_test.shape

(26535, 30)

Se tienen 26535 filas y 30 columnas.

Listamos todos los nombres de las columnas del DataFrame

In [14]:
ds_hoteles_test.columns.tolist()

['hotel',
 'lead_time',
 'arrival_date_year',
 'arrival_date_month',
 'arrival_date_week_number',
 'arrival_date_day_of_month',
 'stays_in_weekend_nights',
 'stays_in_week_nights',
 'adults',
 'children',
 'babies',
 'meal',
 'country',
 'market_segment',
 'distribution_channel',
 'is_repeated_guest',
 'previous_cancellations',
 'previous_bookings_not_canceled',
 'reserved_room_type',
 'assigned_room_type',
 'booking_changes',
 'deposit_type',
 'agent',
 'company',
 'days_in_waiting_list',
 'customer_type',
 'adr',
 'required_car_parking_spaces',
 'total_of_special_requests',
 'id']

A partir de lo anteriormente visto, ahora nos interesa saber qué tipos de datos contienen las columnas del DataFrame

In [16]:
ds_hoteles_test.dtypes

hotel                              object
lead_time                           int64
arrival_date_year                   int64
arrival_date_month                 object
arrival_date_week_number            int64
arrival_date_day_of_month           int64
stays_in_weekend_nights             int64
stays_in_week_nights                int64
adults                              int64
children                          float64
babies                              int64
meal                               object
country                            object
market_segment                     object
distribution_channel               object
is_repeated_guest                   int64
previous_cancellations              int64
previous_bookings_not_canceled      int64
reserved_room_type                 object
assigned_room_type                 object
booking_changes                     int64
deposit_type                       object
agent                             float64
company                           

Nos interesa saber también si hay datos faltantes, para ver si en algún futuro podemos hacer una limpieza

In [17]:
print(ds_hoteles_test.isna().sum())

hotel                                 0
lead_time                             0
arrival_date_year                     0
arrival_date_month                    0
arrival_date_week_number              0
arrival_date_day_of_month             0
stays_in_weekend_nights               0
stays_in_week_nights                  0
adults                                0
children                              0
babies                                0
meal                                  0
country                              95
market_segment                        0
distribution_channel                  0
is_repeated_guest                     0
previous_cancellations                0
previous_bookings_not_canceled        0
reserved_room_type                    0
assigned_room_type                    0
booking_changes                       0
deposit_type                          0
agent                              3363
company                           25218
days_in_waiting_list                  0


Siguiendo la idea anterior, ahora veamos cuántos datos no nulos tenemos por columna

In [18]:
print(ds_hoteles_test.count())

hotel                             26535
lead_time                         26535
arrival_date_year                 26535
arrival_date_month                26535
arrival_date_week_number          26535
arrival_date_day_of_month         26535
stays_in_weekend_nights           26535
stays_in_week_nights              26535
adults                            26535
children                          26535
babies                            26535
meal                              26535
country                           26440
market_segment                    26535
distribution_channel              26535
is_repeated_guest                 26535
previous_cancellations            26535
previous_bookings_not_canceled    26535
reserved_room_type                26535
assigned_room_type                26535
booking_changes                   26535
deposit_type                      26535
agent                             23172
company                            1317
days_in_waiting_list              26535


Comenzamos con la Exploracion Inicial: Vamos a comenzar a analizar cada variable del dataset, considerando ciertos aspectos de interés para nuestro análisis